cloud/pod_util.go

```diff
@@ -335,3 +356,102 @@ func exportPodEnvVars(smConf evergreen.SecretsManagerConfig, opts pod.TaskContai
 
 	return allEnvVars
 }
+
+// GetFilteredResourceIDs gets at most n resources that match the given resource
+// and tag filters. If n is not a positive integer, the results are not limited.
+func GetFilteredResourceIDs(ctx context.Context, c cocoa.TagClient, resources []string, tags map[string][]string, limit int) ([]string, error) {
+	if limit == 0 {
+		return []string{}, nil
+	}
+	if limit < 0 {
+		limit = math.MaxInt64
+	}
+
+	var tagFilters []*resourcegroupstaggingapi.TagFilter
+	for key, vals := range tags {
+		tagFilters = append(tagFilters, &resourcegroupstaggingapi.TagFilter{
+			Key:    aws.String(key),
+			Values: utility.ToStringPtrSlice(vals),
+		})
+	}
+	resourceFilters := utility.ToStringPtrSlice(resources)
+
+	var allIDs []string
+	remaining := limit
+	for ids, nextToken, err := getResourcesPage(ctx, c, resourceFilters, tagFilters, nil, limit); ; ids, nextToken, err = getResourcesPage(ctx, c, resourceFilters, tagFilters, nextToken, remaining) {
+		if err != nil {
+			return nil, errors.Wrap(err, "getting resources matching filters")
+		}
+		allIDs = append(allIDs, ids...)
+		remaining = remaining - len(ids)
+		if remaining <= 0 {
+			break
+		}
+		if nextToken == nil {
+			break
+		}
+	}
+
+	return allIDs, nil
+}
+
+func getResourcesPage(ctx context.Context, c cocoa.TagClient, resourceFilters []*string, tagFilters []*resourcegroupstaggingapi.TagFilter, nextToken *string, limit int) ([]string, *string, error) {
+	var ids []string
+
+	resp, err := c.GetResources(ctx, &resourcegroupstaggingapi.GetResourcesInput{
+		PaginationToken:     nextToken,
```

Comment:
There is a `ResourcesPerPage` property you can set here from 1 to 100, which might be relevant if limit is < 100, and might reduce the amount of calls we need to make. 

The docs do say that "A query can return fewer than this maximum, even if there are more results still to return. You should always check the PaginationToken response value to see if there are more results", so it isn't an airtight thing.  Just letting you know it's an option in case you feel like it'd be helpful.

Reply:
Hmm I don't have that strong of feelings one way or the other on using vs. not using the page size. Ideally, we just want it to return as many as possible per page (which is what it does by default), so the number of calls needed is the same whether we set page size or not. We don't care too deeply about AWS returning too many results in a single page either, since AWS is supposedly structured to be efficient at getting tagged resources.

Reply:
Yeah, makes sense. Looking closer it doesn't really seem like there's much reason to use it especially if there isn't a guarantee returning less than the limit means we are done getting results.